BuildingBlocks/test/BuildingBlocks.Application.Tests/Mediatr/QuotaEnforcerBehaviorTests.cs

```diff
@@ -1,100 +1,109 @@
 ﻿using Enmeshed.BuildingBlocks.Application.Abstractions.Exceptions;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.UserContext;
 using Enmeshed.BuildingBlocks.Application.Attributes;
 using Enmeshed.BuildingBlocks.Application.MediatR;
 using Enmeshed.BuildingBlocks.Domain;
 using FluentAssertions;
 using MediatR;
 using Xunit;
 
+using DomainMetricStatus = Enmeshed.BuildingBlocks.Domain.MetricStatus;
+
 namespace Enmeshed.BuildingBlocks.Application.Tests.Mediatr;
 public class QuotaEnforcerBehaviorTests
 {
-    public QuotaEnforcerBehaviorTests() { }
+    private readonly IUserContext _userContextStub;
+
+    public QuotaEnforcerBehaviorTests() {
+        _userContextStub = new UserContextStub();
+    }
 
-    [ApplyQuotasForMetrics("MetricKey")]
-    private class TestCommand : IRequest<IResponse> { }
+    [ApplyQuotasForMetrics("*")]
+    private class TestCommand : IRequest { }
 
     private class IResponse { }
 
     [Fact]
-    public async void Throws_QuotaExhaustedException_OnExhaustedQuota()
+    public async void Throws_QuotaExhaustedException_when_at_least_one_metric_is_exhausted()
     {
         // Arrange
         var metricStatusesStubRepository = new MetricStatusesStubRepository(new() {
-            new MetricStatus(new MetricKey("KeyOne"), DateTime.UtcNow.AddDays(1)) 
+            TestData.MetricStatus.ThatIsExhaustedFor1Day, TestData.MetricStatus.ThatWasExhaustedUntilYesterday
```

Comment:
Nice test case. But I think this should be a separate one that is called "Throws_QuotaExhaustedException_with_exhaustion_date_when_multiple_metrics_are_exhausted"